craft_parts/utils/partition_utils.py

```diff
@@ -51,8 +59,81 @@ def validate_partition_names(partitions: Optional[Sequence[str]]) -> None:
     if len(partitions) != len(set(partitions)):
         raise errors.FeatureError("Partitions must be unique.")
 
+    _validate_partition_names(partitions)
+
+    _validate_namespace_conflicts(partitions)
+
+
+def _is_valid_partition(partition: str) -> bool:
+    """Check if a partition name is valid.
+
+    :param partition: partition to check
+
+    :returns: true if the namespaced partition is valid
+    """
+    return bool(re.fullmatch("[a-z]+", partition))
```

Comment:
If we compile the regex at import time and make it a constant:

```suggestion
_VALID_PARTITION_REGEX = re.compile("[a-z]+", re.ASCII)

def _is_valid_partition(partition: str) -> bool:
    """Check if a partition name is valid.

    :param partition: partition to check

    :returns: true if the namespaced partition is valid
    """
    return bool(_VALID_PARTITION_REGEX.fullmatch(partition))
```

We can make it faster for repeated use (and optionally do direct testing of the regex using hypothesis.)

Reply:
In this specific case `name.isalpha() and name.islower()` could be more efficient but at this scale the difference would be negligible.